cmd/state-tray/internal/open/open_lin.go

```diff
@@ -28,3 +28,38 @@ func Prompt(command string) error {
 
 	return nil
 }
+
+func preferredShellWithFallback(fallback string) (string, error) {
+	currentUser, err := user.Current()
+	if err != nil {
+		return "", locale.WrapError(err, "err_user_unknown", "Cannot get current user")
+	}
+
+	f, err := os.Open("/etc/passwd")
+	if err != nil {
+		return "", locale.WrapError(err, "err_open_passwd", "Cannot open passwd file")
+	}
+	defer f.Close()
+
+	prefix := currentUser.Name + ":"
+	var shell string
+
+	sc := bufio.NewScanner(f)
+	for sc.Scan() {
+		line := sc.Text()
+		if !strings.HasPrefix(line, prefix) {
+			continue
+		}
+		parts := strings.Split(line, ":")
+		shell = parts[len(parts)-1]
+	}
+	if err := sc.Err(); err != nil {
+		return "", locale.WrapError(err, "err_scan_passwd", "/etc/passwd file scan failed")
```

Comment:
Here I would also return the fallback.  Maybe this function should not return an error at all?  Just log problems via `logging.Debug`...

Reply:
From mobile: This looks like the code reviewed is from an older commit (or I didn't push!). Does this still apply?

Reply:
Yes, I think so.  I believe that we should always use the fallback from this function if an error occurs.  Never forward the error, even if it looks less sketchy. :)

Once you add rollbar logging, you can still log the errors there...